core/src/main/scala/org/apache/spark/ui/DriverLogPage.scala

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.ui
+
+import javax.servlet.http.HttpServletRequest
+
+import scala.xml.{Node, Unparsed}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.Logging
+import org.apache.spark.internal.config.DRIVER_LOG_LOCAL_DIR
+import org.apache.spark.util.Utils
+import org.apache.spark.util.logging.DriverLogger.DRIVER_LOG_FILE
+import org.apache.spark.util.logging.RollingFileAppender
+
+/**
+ * Live Spark Driver Log UI Page.
+ *
+ * This is similar with Spark worker's LogPage class.
+ */
+private[ui] class DriverLogPage(
+    parent: DriverLogTab,
+    conf: SparkConf)
+  extends WebUIPage("") with Logging {
+  require(conf.get(DRIVER_LOG_LOCAL_DIR).nonEmpty, s"Please specify ${DRIVER_LOG_LOCAL_DIR.key}.")
+
+  private val supportedLogTypes = Set(DRIVER_LOG_FILE, "stderr", "stdout")
```

Comment:
Will it use stderr and stdout? Looks like `DriverLogger` won't write to these two?

Reply:
Yes, for now, `stdout` and `stderr` is not used in this code path, but REST API will work in the same way. I keep them to keep this `DriverLogPage.scala` to keep similar like `LogPage.scala` which uses `stdout` and `stderr`. In addition, for the future extensions which points to log4j2 configured directory.